src/globus_cli/commands/gcs/endpoint/role/create.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+import typing as t
+import uuid
+
+import click
+import globus_sdk
+
+from globus_cli.commands.gcs.endpoint.role._common import role_fields
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command, endpoint_id_arg
+from globus_cli.termio import TextMode, display
+from globus_cli.utils import resolve_principal_urn
+
+_VALID_ROLES = ["administrator", "activity_manager", "activity_monitor"]
+
+
+@command(
+    "show",
+)
+@endpoint_id_arg
+@click.argument("ROLE", type=click.Choice(_VALID_ROLES), metavar="ROLE")
+@click.argument("PRINCIPAL", type=str)
+@click.option(
+    "--principal-type",
+    type=click.Choice(["identity", "group"]),
+    help="Qualifier to specify what type of principal (identity or group) is provided.",
+)
+@LoginManager.requires_login("transfer")
+def create_command(
+    login_manager: LoginManager,
+    *,
+    endpoint_id: uuid.UUID,
+    role: t.Literal["administrator", "activity_manager", "activity_monitor"],
+    principal: str,
+    principal_type: t.Literal["identity", "group"] | None,
+) -> None:
+    """
+    Create a role on a GCS Endpoint.
+
+    ROLE may be one of: ["administrator", "activity_manager", "activity_monitor"].
+
+    PRINCIPAL may be either a username or a UUID/URN for an identity or group.
+
+       * If a UUID, you must also provide --principal-type.
```

Comment:
I would make some pretty minor adjustments to this text on three points:
- "may" -> "must", "is"
- unwrap that list of roles by removing the brackets and adding `and`
- correct `--principal-type` to note the default

```suggestion
    ROLE must be one "administrator", "activity_manager", or "activity_monitor".

    PRINCIPAL is either a username or a UUID/URN for an identity or group.
    If it is a UUID, use `--principal-type` to provide the type (defaults to "identity").
```

That third point is the most important. Regarding the other changes, I'm really just polishing things a little. I think we already throw a lot of string syntaxes at users (e.g. `{foo,bar}` vs `[foo|bar]` vs ...) and I want to try to reduce the degree of that.

Reply:
Sure I can make these. You may be able to tell but I wrote this docstring while UUIDs didn't assume that principal type was identity & apparently forgot to come back and update it.